build-system/common/git.js

```diff
@@ -20,39 +20,19 @@
  */
 
 const {
-  isTravisBuild,
   isTravisPullRequestBuild,
   travisPullRequestBranch,
   travisPullRequestSha,
 } = require('./travis');
 const {getStdout} = require('./exec');
 
 /**
- * Returns the commit at which the current branch was forked off of master.
- * On Travis, there is an additional merge commit, so we must pick the first of
- * the boundary commits (prefixed with a -) returned by git rev-list.
- * On local branches, this is merge base of the current branch off of master.
+ * Returns the remote tracking branch associated with the local master branch.
  * @return {string}
  */
-function gitBranchCreationPoint() {
-  if (isTravisBuild()) {
-    const traviPrSha = travisPullRequestSha();
-    return getStdout(
-      `git rev-list --boundary ${traviPrSha}...master | grep "^-" | head -n 1 | cut -c2-`
-    ).trim();
-  }
-  return gitMergeBaseLocalMaster();
-}
-
-/**
- * Returns the `master` parent of the merge commit (current HEAD) on Travis.
- * Note: This is not the same as origin/master (a moving target), since new
- * commits can be merged while a Travis build is in progress.
- * See https://travis-ci.community/t/origin-master-moving-forward-between-build-stages/4189/6
- * @return {string}
- */
-function gitTravisMasterBaseline() {
-  return getStdout('git merge-base origin/master HEAD').trim();
+function gitUpstreamMaster() {
+  const upstreamName = getStdout('git config branch.master.remote').trim();
```

Comment:
If one were to follow the instructions in our [development docs](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#set-up-aliases-for-the-remote-git-repositories) while setting up repo aliases, they'd end up with something like the following:

![image](https://user-images.githubusercontent.com/26553114/80331107-271b1b00-8815-11ea-949d-b25833d8617e.png)

In the example above:

- `origin/master` refers to the `master` branch on my fork `rsimha/amphtml` (not guaranteed to be up to date)
- `upstream/master` refers to the `master` branch on the main `ampproject/amphtml` repo (always up to date)
- The result of `git config branch.master.remote` returns `origin`, not `upstream`

Would it make sense to update the logic for `upstreamName` as follows?
- Check for the existence of `upstream/master` and use it if it exists
- If not, use `(getStdout('git config branch.master.remote').trim() || 'origin') + '/master'`

Reply:
PR updated with more rigorous detection of upstream remote name. Thanks for pointing out this issue. I didn't realize the dev docs suggest tracking your own repo instead of upstream. Marking resolved. Please comment on updated code if there are any issues.